lib/segment/src/index/field_index/binary_index.rs

```diff
@@ -218,6 +219,22 @@ impl BinaryIndex {
     pub fn values_has_false(&self, point_id: PointOffsetType) -> bool {
         self.memory.get(point_id).has_false()
     }
+
+    pub fn iter_values_map(&self) -> impl Iterator<Item = (bool, IdIter<'_>)> + '_ {
+        vec![
+            (false, Box::new(self.memory.iter_has_false()) as IdIter),
+            (true, Box::new(self.memory.iter_has_true()) as IdIter),
+        ]
+        .into_iter()
+    }
+
+    pub fn iter_counts_per_value(&self) -> impl Iterator<Item = (bool, usize)> + '_ {
+        vec![
+            (false, self.memory.falses_count()),
+            (true, self.memory.trues_count()),
+        ]
+        .into_iter()
```

Comment:
Do we need to filter items with a count of 0 here?

Reply:
It is already filtered later :)